sdk/cosmos/sdk/src/main/java/com/azure/data/cosmos/sync/CosmosSyncClient.java

```diff
@@ -0,0 +1,230 @@
+    }
+
+    /**
+     * Creates a database.
+     *
+     * @param id the id of the database
+     * @return the {@link CosmosSyncDatabaseResponse} with the created database.
+     * @throws CosmosClientException the cosmos client exception.
+     */
+    public CosmosSyncDatabaseResponse createDatabase(String id) throws CosmosClientException {
+        return mapDatabaseResponseAndBlock(asyncClientWrapper.createDatabase(id));
+
+    }
+
+    /**
+     * Creates a database.
+     *
+     * @param databaseProperties {@link CosmosDatabaseProperties} the database properties.
+     * @param throughput the throughput
+     * @param options {@link CosmosDatabaseRequestOptions} the request options
+     * @return the {@link CosmosSyncDatabaseResponse} with the created database.
+     * @throws CosmosClientException the cosmos client exception
+     */
+    public CosmosSyncDatabaseResponse createDatabase(CosmosDatabaseProperties databaseProperties,
+                                                     int throughput,
+                                                     CosmosDatabaseRequestOptions options) throws CosmosClientException {
+        return mapDatabaseResponseAndBlock(asyncClientWrapper.createDatabase(databaseProperties, throughput, options));
+    }
+
+    /**
+     * Creates a database.
+     *
+     * @param databaseProperties {@link CosmosDatabaseProperties} the database properties.
+     * @param throughput the throughput
+     * @return the {@link CosmosSyncDatabaseResponse} with the created database.
+     * @throws CosmosClientException the cosmos client exception
+     */
+    public CosmosSyncDatabaseResponse createDatabase(CosmosDatabaseProperties databaseProperties,
+                                                     int throughput) throws CosmosClientException {
+        return mapDatabaseResponseAndBlock(asyncClientWrapper.createDatabase(databaseProperties, throughput));
+    }
+
+
+    /**
+     * Creates a database.
+     *
+     * @param id the id of the database
+     * @param throughput the throughput
+     * @return the {@link CosmosSyncDatabaseResponse} with the created database.
+     * @throws CosmosClientException the cosmos client exception
+     */
+    public CosmosSyncDatabaseResponse createDatabase(String id, int throughput) throws CosmosClientException {
+        return mapDatabaseResponseAndBlock(asyncClientWrapper.createDatabase(id, throughput));
+    }
+
+    CosmosSyncDatabaseResponse mapDatabaseResponseAndBlock(Mono<CosmosDatabaseResponse> databaseMono)
+            throws CosmosClientException {
+        try {
+            return databaseMono
+                           .map(this::convertResponse)
+                           .block();
+        } catch (Exception ex) {
+            final Throwable throwable = Exceptions.unwrap(ex);
+            if (throwable instanceof CosmosClientException) {
+                throw (CosmosClientException) throwable;
```

Comment:
nice way of keeping the original exception and ensuring we through correct exception type. great job Bhaskar for taking care of this.

Reply:
Thankyou